tests/bluetooth/shell/audio.conf

```diff
@@ -216,3 +216,23 @@ CONFIG_BT_CAP_COMMANDER_LOG_LEVEL_DBG=y
 CONFIG_BT_CAP_COMMON_LOG_LEVEL_DBG=y
 CONFIG_BT_TMAP_LOG_LEVEL_DBG=y
 CONFIG_BT_GMAP_LOG_LEVEL_DBG=y
+
+# Controller settings
+CONFIG_BT_CTLR_DATA_LENGTH_MAX=251
+
+# Support having both a connectable and a non-connectable set
+CONFIG_BT_CTLR_ADV_SET=2
+CONFIG_BT_CTLR_ADV_AUX_SET=2
+CONFIG_BT_CTLR_ADV_SYNC_SET=2
+CONFIG_BT_CTLR_ADV_DATA_LEN_MAX=191
+CONFIG_BT_CTLR_ADV_DATA_BUF_MAX=6
+
+CONFIG_BT_CTLR_SCAN_DATA_LEN_MAX=1650
+
+# Match the number of broadcast streams supported in BAP
+CONFIG_BT_CTLR_SYNC_ISO_STREAM_MAX=4
+CONFIG_BT_CTLR_ADV_ISO_STREAM_MAX=4
+
+# Match the number of unicast streams supported in BAP
+CONFIG_BT_CTLR_ISOAL_SOURCES=2
+CONFIG_BT_CTLR_ISOAL_SINKS=2
```

Comment:
Should we instead have overlay conf (say, `overlay-bt_ll_sw_split.conf` or `overlay-some_other_controller.conf`; otherwise these would throw Kconfig warnings on boards (etc. nrf5340dk_nrf5340_cpuapp) without `CONFIG_BT_CTLR` enabled.

Reply:
That might be a better solution, yes. 

The reason why I went with this, is that a few other samples and tests do this too (without the overlay). 

Generally not a huge fan of overlays as they make it harder for people (especially newcomers) to just do `west build` and `west flash` for things to work. 

Also did consider the approach of entering this into to board config files (which are automatically applied), but would then have to find a suitable selection of boards to apply this for (issue was originally found for the nrf52_bsim board). 

A bit of a shame that we cannot do 

```
if CONFIG_BT_CTLR
<apply these settings>
```

Reply:
I agree with you. But to support selection of controller choice for a board, and/or to have application specific controller features, at this point overlays seems the only known option for me.

Reply:
> at this point overlays seems the only known option for me.

I wonder if there are some way we can automatically apply the overlay if `CONFIG_BT_CTLR` then :o 

@aescolar usually have some good ideas for this stuff :D

Reply:
> I wonder if there are some way we can automatically apply the overlay if CONFIG_BT_CTLR then :o

One option is defining a kconfig file for the test app itself. There you can add to or override any zephyr kconfig option with any `if` you want

Reply:
Yeah, that is another option, but that's done by modifying the `default` value of the Kconfig options, right? I really dislike that approach :( 

So I think the two best options are
1) Current suggestion where we get warnings for non-controller builds
2) Overlay which require users to know that they must include the overlay when building

and with additional options to
3) Override defaults of Kconfig in the shell Kconfig file
4) Add the controller options on a per-board approach

None are (IMO) very good :(

Reply:
i am ok for this to be merged as-is (the discussion is here in this PR for future reference). When other vendors use this shell application, they can move theses Kconfig assignments out to overlays as necessary.